### ISSUE TITLE
Microsoft.Extensions.Configuration.UserSecrets 6.0.0-preview.1.21102.12 throwing secrets.json error in CI/CD pipelines

### ISSUE THREAD

**Issue by samsmithnz** [dotnet/runtime#48485]
### Description

User Secrets error is being generated in a CI/CD pipeline, when secrets.json file shouldn't be expected.

Steps: 
1. Create .NET 5 project
1. Added user secrets. 
2. Code runs locally and in CI/CD pipelines.
3. Upgrade to .NET 6 project (and preview NuGet 6.* packages)
3. Code runs locally, but fails in CI/CD pipelines, with error:

"The configuration file 'secrets.json' was not found and is not optional."

Expected:
- Code runs without the secrets.json file being present

### Configuration

.NET 6, Microsoft.Extensions.Configuration.UserSecrets: 6.0.0-preview.1.21102.12

### Regression?

This works in .NET 5, Microsoft.Extensions.Configuration.UserSecrets: 5.0.0.*

I created a repo that demos this: https://github.com/samsmithnz/UserSecretsRegression
- Main branch is .NET 5 and works.
- dotnet6Upgrade branch has a pull request that fails with the message below (https://github.com/samsmithnz/UserSecretsRegression/pull/1)

### Other information

Stack trace:
```
System.IO.FileNotFoundException: The configuration file 'secrets.json' was not found and is not optional. The physical path is '/home/runner/work/UserSecretsRegression/UserSecretsRegression/UserSecrets/UserSecrets.Tests/bin/Release/net6.0/secrets.json'.
  Stack Trace:
      at Microsoft.Extensions.Configuration.FileConfigurationProvider.HandleException(ExceptionDispatchInfo info)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load()
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at UserSecrets.Tests.UnitTest1.TestMethod1() in /home/runner/work/UserSecretsRegression/UserSecretsRegression/UserSecrets/UserSecrets.Tests/UnitTest1.cs:line 13
```

**Comment by samsmithnz** [comment]
For anyone else blocked by this, I created a workaround with some PowerShell to create an empty secrets.json file while we wait for this to be looked at. This is a GitHub action, but the equivalent would work on any system.
```
    - run: |
        #Create the secrets.json file
        New-Item UserSecrets\UserSecrets.Tests\secrets.json
        #Insert in empty braces to make it a valid (empty) json file
        "{}" | Set-Content UserSecrets\UserSecrets.Tests\secrets.json
        #Copy the secrets.json file to the bin directory, so that it will be present for the tests 
        xcopy "${{ github.workspace }}\UserSecrets\UserSecrets.Tests\secrets.json" "${{ github.workspace }}\UserSecrets\UserSecrets.Tests\bin\release\net6.0\"  /f /y
      shell: powershell
```

I created a new pull request to prove this out, here: https://github.com/samsmithnz/UserSecretsRegression/pull/2

**Comment by maryamariyan** [comment]
This behavioral change was a fix introduced in 6.0 for Issue https://github.com/dotnet/runtime/issues/44150.

Adding user secrets sets the optional bool flag as false and we should throw an exception (which wasnt happening before). If you wish to make it optional you need to pass the flag yourself in https://github.com/samsmithnz/UserSecretsRegression/blob/2c1cda555cde2a4b14ac3b968ed2044bb16181c4/UserSecrets/UserSecrets.Tests/UnitTest1.cs#L14 by passing true.

**Comment by samsmithnz** [comment]
@maryamariyan: This is an odd one. After reading the notes, I understand the scenario and why you've added an exception, but does this make sense for CI/CD situations and secret management in general? 

I definitely don't want to commit/push my secrets.json file to my Git repo. Do you recommend I add a parameter to my appsettings.json file to pass in the false when building with CI/CD, and leave it to true when debugging/running locally?

**Comment by maryamariyan** [comment]
The default value for `optional` is false. If you would like to keep the old behavior, (to keep optional behavior) then you need to specifically set optional to true in [AddJsonFile](https://apisof.net/catalog/84529c3b-1071-5f6a-64d2-4f2ff58a9e0b).

**Comment by samsmithnz** [comment]
Thanks I did this and it works: https://github.com/samsmithnz/UserSecretsRegression/pull/3

I'm going to leave the repo around for when others run into this, as I feel like we've broken a lot of code in a subtle way (and the default should be true) :).  

Thanks for your patience and input! Appreciate it.
